Lab philosophy
==============

The lab motto is "Better science through superior software" - we
believe that, as biology becomes increasingly reliant on computation
and data analysis, better software is needed to progress biological
investigation.  This includes better software development practices
and workflows, placing an emphasis on core scientific values of open
source and reproducibility, and an interest in developing flexible
libraries and reusable/remixable software.

The guiding principle of the lab is to advance the biological sciences
through enhanced use, extension, and development of computational
tools and approaches.  Our primary focus for now is on "-omics" data -
primarily genomic, transcriptomic, and metagenomic sequencing data.

Open science
------------

To the best of our ability, we practice open science; **the default is
set to "open"**.

* all software should be available under a BSD open source license.

* most papers are written openly and submitted to preprint servers.

* all data is made available as soon as practical.

* collaboration is good.

We also encourage blogging and open discussion of research on social
media.

Lab details
-----------

Within the lab, we try to practice `lazy consensus
<http://nowviskie.org/2012/lazy-consensus/>`__.

We have a code of conduct; please see :doc:`coc`.

All software within the lab should be written in C++ and/or Python, in
order to maximize reuse and remixability within the lab.

Miscellany
----------

Other miscellaneous thoughts --

* postdocs are generally welcome to take projects with them and
  collaborate on lab projects after they leave the lab, although Titus
  may have suggestions to help with career considerations.

* given finite resources, not all problems are worth solving.  Choose
  problems worthy of your time and energy.

* teaching and training are core values within the lab.  However, please
  be mindful of the need to balance teaching/training with research efforts.
